.github/workflows/build.yaml

```diff
@@ -378,14 +378,22 @@ jobs:
       - name: Display ccache stats
         run: ccache -s
 
+      - name: enable asserts on asan build
+        if: matrix.runtimeCheck == 'asan' || matrix.runtimeCheck == 'OFF'
+        run: echo "DispatchCMakeAsserts=ON" >> $GITHUB_ENV
+
+      - name: disable asserts on tsan build
+        if: matrix.runtimeCheck == 'tsan'
+        run: echo "DispatchCMakeAsserts=OFF" >> $GITHUB_ENV
```

Comment:
This does not work. Debug builds always have asserts enabled; `-DQD_ENABLE_ASSERTIONS` is only a way to turn them on where they would be otherwise disabled.

Notice that on Debug builds, the option does not undefine anything (which would be necessary to disable asserts):

https://github.com/apache/qpid-dispatch/blob/main/CMakeLists.txt#L218